rse-project.Rmd

```diff
@@ -21,7 +21,214 @@ small pieces in predictable places with readable names are easier to find and us
+
+> "Like deciding when a chunk of code should be made a function, the ultimate goal of dividing research into distinct projects is to help you and others best understand your work. Some researchers create a separate project for each manuscript they are working on, while others group all research on a common theme, data set, or algorithm into a single project."
+
+> "As a rule of thumb, divide work into projects based on the overlap in data and code files. If 2 research efforts share no data or code, they will probably be easiest to manage independently. If they share more than half of their data and code, they are probably best managed together, while if you are building tools that are used in several projects, the common code should probably be in a project of its own." 
+
+Like features (Chapter \@ref(branches)),
+what exactly constitutes a "project" requires a bit of judgment,
+and different people will make different decisions.
+Some common criteria are one project per publication,
+one project per deliverable piece of software,
+or one project per team.
+The first tends to be too small:
+a good data set will result in several reports,
+and the goal of some projects is to produce a steady stream of reports (such as monthly forecasts).
+The second is a good fit for software engineering projects
+whose primary aim is to produce tools rather than results,
+but is inappropriate for most data analysis work.
+The third tends to be too large:
+a team of half a dozen people may work on many different things at once,
+and a repository that holds them all quickly looks like someone's basement.
+
+The best rule of thumb we have found for deciding what is and isn't a project
+is to ask what you have meetings about.
+If the same set of people need to get together on a regular basis to talk about something,
+that "something" probably deserves its own repository.
+And if the list of people changes slowly over time but the meetings continue,
+that's an even stronger sign.
+
+## What files should every project contain? {#project-boilerplate}
+
+TODO: What are the bare minimums here?  I'd say must have `README`, even for your own projects that only live locally.  As soon as something is on GitHub must have `LICENSE` and `CONDUCT`?
+
+Most projects' repositories contain a few files that weren't mentioned in Noble's paper,
+but which have become conventional in open source projects.
+All of these files may be plain text or Markdown,
+and may have a `.txt` or `.md` suffix (or no suffix at all),
+but please use the principal names given,
+in all caps,
+since a growing number of tools expect them.
+
+-   `README`:
+    the project's title and a one-paragraph description of its purpose or content.
+    This file is displayed by GitHub Pages as the project's home page.
+
+-   `LICENSE`:
+    the project's license (discussed in Chapter \@ref(inclusive)).
+
+-   `CONDUCT`:
+    its code of conduct
+    (also discussed in Chapter \@ref(inclusive)).
+
+-   `CITATION`:
```

Comment:
I would add "contributing" to this list to help guide and welcome contributors. https://mozilla.github.io/open-leadership-training-series/articles/building-communities-of-contributors/write-contributor-guidelines/

Reply:
I think another important document is a roadmap. This could be part of contributing, the readme, or separate, and doesn't have to be detailed, but give newcomers and idea of where the project is headed. https://mozilla.github.io/open-leadership-training-series/articles/opening-your-project/start-your-project-roadmap/

Reply:
Great idea!